File: hamster-back/library-core/src/main/java/com/librarycore/book/domain/BookInventory.java
package com.librarycore.book.domain;

import identity.BookInventoryId;
import identity.BookSkuId;

public record BookInventory(
        BookInventoryId bookInventoryId,
        BookSkuId bookSkuId,
        BookStatus status
) {

    public BookInventory loaned() {
        return new BookInventory(bookInventoryId, bookSkuId, BookStatus.LOANED);
    }

    public BookInventory available() {
        return new BookInventory(bookInventoryId, bookSkuId, BookStatus.AVAILABLE);
    }
}

File: hamster-back/library-core/src/main/kotlin/com/librarycore/book/domain/BookInventory.kt
package com.librarycore.book.domain

import identity.BookInventoryId
import identity.BookSkuId

data class BookInventory(
    val bookInventoryId: BookInventoryId?,
    val bookSkuId: BookSkuId?,
    val status: BookStatus?
) {
    fun loaned(): BookInventory {
        return BookInventory(bookInventoryId, bookSkuId, BookStatus.LOANED)
    }

    fun available(): BookInventory {
        return BookInventory(bookInventoryId, bookSkuId, BookStatus.AVAILABLE)
    }
}